### PR TITLE
tools/scylla-sstable: pass error handler to utils::config_file::read_from_file()

### PR DESCRIPTION
**Pull request by denesb**
The default error handler throws an exception, which means scylla-sstable will exit with exception if there is any problem in the configuration. Not even ScyllaDB itself is this harsh -- it will just log a warning for most errors. A tool should be much more lenient. So this patch passes an error handler which just logs all errors with debug level.
If reading an sstable fails, the user is expected to investigate turning debug-level logging on. When they do so, they will see any problems while reading the configuration (if it is relevant, e.g. when using EAR).

Fixes: #16538

### CODE REVIEW

**Comment by denesb** [comment]
v2:
* Use with scylla_sstable to produce schema_file and sstable in the test
* New patch to allow passing --scylla-yaml-file along with --schema-file

File: tools/scylla-sstable.cc [hunk 1]
```diff
@@ -3013,7 +3013,9 @@ Validate the specified sstables:
             if (!schema_sources) {
                 sst_log.debug("No user-provided schema source, attempting to auto-detect it");
                 schema_with_source = try_load_schema_autodetect(app_config, dbcfg);
-            } else if (schema_sources == 1) {
+            } else if (schema_sources == 1 || (schema_sources == 2 && app_config.contains("scylla-yaml-file"))) {
+                // We make an exception for the case where 2 schema sources are provided, but one of them is scylla-yaml file.
+                // We want to always accept the --scylla-yaml-file option.
```
Comment (tchaikov) [inline tools/scylla-sstable.cc:3020]: could you help me understand what "e.v." stands for in the commit message?

File: tools/scylla-sstable.cc [hunk 2]
```diff
@@ -3013,7 +3013,9 @@ Validate the specified sstables:
...
                 sst_log.debug("Single schema source provided");
```
Comment (tchaikov) [inline tools/scylla-sstable.cc:3021]: nit, the logging message might not be accurate anymore.

File: tools/scylla-sstable.cc (hunk 1 above)
Comment (denesb) [reply tools/scylla-sstable.cc:3020]: It is supposed to be "e.g." Will fix.

File: tools/scylla-sstable.cc (hunk 2 above)
Comment (denesb) [reply tools/scylla-sstable.cc:3021]: It still is (conceptually). I will follow-up with erasing this single-schema source requirement, and just document the option evaluation order instead. But this PR is a bugfix, which will be backported (in enterprise), so I want to keep this change out of it.

**Comment by denesb** [comment]
v1.1:
* Fix typo in commit message (s/e.v./e.g./)

File: tools/scylla-sstable.cc (hunk 2 above)
Comment (tchaikov) [reply tools/scylla-sstable.cc:3021]: ack.

**Review by tchaikov — APPROVED** [review]
lgtm

**Pull request merged**